src/viam/sdk/robot/client.hpp

```diff
@@ -23,13 +25,62 @@ namespace viam {
 namespace sdk {
 
 using grpc::Channel;
-using viam::common::v1::ResourceName;
-using viam::common::v1::Transform;
-using viam::robot::v1::FrameSystemConfig;
 using viam::robot::v1::RobotService;
```

Comment:
I wasn't really sure what to do with this.

Reply:
Because of `std::unique_ptr<RobotService::Stub> stub_` in `RobotClient`? If so, I have a few suggestions:
 
- You could introduce a `Robot` base class. Then `RobotClient` would implement it and the factory functions would return a `shared_ptr<Robot>` but it would point to an instance of `RobotClient`.
- You could use the "pimpl" idiom to make a little opaque holder for the `Stub`:

In `client.hpp`:
```
class RobotClient {
  ...
private:
    struct Impl;
    unique_ptr<Impl> impl_;
};
```

Then in `client.cpp`:
```
struct RobotClient::Impl {
    Impl(RobotService::Stub stub) : stub_(std::move(stub)) {}
    RobotService::Stub stub_;
};

...

class RobotClient::RobotClient(...) : ..., impl_(std::make_unique<Impl>(RobotService::NewStub(channel_)), ... {
  ...
}

```

Unless @stuqdog wants to introduce a `Robot` base class, I think I'd follow the second suggestion for now. It can always be revisited.

Reply:
I think getting rid of this `RobotService` include involves moving much of the logic here and is probably outside the scope of this PR.

Reply:
I get this error: 
```
note: candidate constructor (the implicit copy constructor) not viable: no known conversion from 'std::unique_ptr<viam::robot::v1::RobotService::Stub>' to 'const viam::sdk::RobotClient::Impl' for 1st argument
struct RobotClient::Impl {
```
I had to make the stub a pointer if we wanted to use it with `RobotService::NewStub()`.
```
struct RobotClient::Impl {
    Impl(std::unique_ptr<RobotService::Stub> stub) : stub_(std::move(stub)) {}
    std::unique_ptr<RobotService::Stub> stub_;
};
```
Even so, with this change, `test_robot.cpp` is failing.

Reply:
Let's chat offline about this. I'm surprised to hear tests are failing, I suspect there's probably something small to fix it but it'll be a lot easier to find with synchronous communication.

Reply:
Ah, well if `RobotService::Stub` is non-copyable you can probalby just require an rvalue reference in the `Impl` ctor:

```
struct RobotClient::Impl {
    Impl(RobotService::Stub&& stub) : stub_(std::move(stub)) {}
    RobotService::Stub stub_;
};
```

If that still doesn't work for some reason then yes you will need the additional `unique_ptr`.

Reply:
I believe you can now remove `using viam::robot::v1::RobotService;` from this header.